[WPE] Build failure with Clang 4.0.1: no matching conversion for functional-style cast from 'pointer' (aka 'unsigned short *') to 'WTF::String'
https://bugs.webkit.org/show_bug.cgi?id=175477

Reviewed by Žan Doberšek.

* Shared/wpe/WebEventFactory.cpp:
  (WebKit::singleCharacterStringForKeyEvent): Cast to UChar* to make
  compiler choose the correct WTF::String constructor.

(revision 223544)

--- webkitgtk/Source/WebKit2/Shared/wpe/WebEventFactory.cpp.orig   2017-12-17 20:37:47.918238974 +0100
+++ webkitgtk/Source/WebKit2/Shared/wpe/WebEventFactory.cpp        2017-12-17 20:39:10.970532449 +0100
@@ -60,5 +60,5 @@
     GUniquePtr<gunichar2> uchar16(g_ucs4_to_utf16(&event->unicode, 1, 0, &length, nullptr));
     if (uchar16)
-        return String(uchar16.get());
+        return String(reinterpret_cast<UChar*>(uchar16.get()), length);
     return String();
 }
